core/src/conn.rs

```diff
@@ -17,19 +17,135 @@ pub struct MongoConnection {
 }
 
 impl MongoConnection {
-    // Creates a new MongoConnection with the given settings and execute a dummy aggregation to
-    // validate it works.
-    // The operation will timeout if it takes more than loginTimeout seconds.
+    // Creates a new MongoConnection with the given settings and lists databases to make sure the
+    // connection is legit.
+    //
+    // The operation will timeout if it takes more than loginTimeout seconds. This timeout is
+    // delegated to the mongo rust driver.
+    //
     // The initial current database if provided should come from SQL_ATTR_CURRENT_CATALOG
     // and will take precedence over the database setting specified in the uri if any.
-    // The initial operation time if provided should come from  and will take precedence over the
+    // The initial operation time if provided should come from and will take precedence over the
     // setting specified in the uri if any.
     pub fn connect(
-        _uri: &str,
-        _current_db: Option<&str>,
-        _operation_timeout: Option<i32>,
-        _login_timeout: Option<i32>,
+        uri: &str,
+        current_db: Option<String>,
+        operation_timeout: Option<i32>,
+        login_timeout: Option<i32>,
     ) -> Result<Self> {
-        unimplemented!()
+        dbg!(uri);
```

Comment:
These `dbg` statements will be removed before merge!

Reply:
I see. Do we have special plan for extra debug logging or we will just rely on SQLTrace to start with?
I think SQLTrace is fine to start with. For adding more logs, I am not sure what would be best. Maybe something to look into in the design?

Reply:
I think SQLTrace is fine to start. We can  consider adding a logging library, but there is always the question of where it will log to (especially on windows).